gdnative-core/src/core_types/access.rs

```diff
@@ -30,17 +30,28 @@ where
 }
 
 /// Trait for array access guards
-#[doc(hidden)]
+///
+/// This trait is sealed.
 #[allow(clippy::len_without_is_empty)]
-pub unsafe trait Guard: Drop {
+pub unsafe trait Guard: Drop + private::Sealed {
     type Target;
     fn len(&self) -> usize;
     fn read_ptr(&self) -> *const Self::Target;
```

Comment:
These items should be `#[doc(hidden)]` I think?

Reply:
Ah, I misunderstood. Will update! That's why it'd be `empty impl` from the public view of the docs.